tools/llvm2bpl/llvm2bpl.cpp

```diff
@@ -122,6 +123,7 @@ int main(int argc, char **argv) {
   pass_manager.add(new llvm::MergeArrayGEP());
   // pass_manager.add(new smack::SimplifyLibCalls());
   pass_manager.add(new llvm::Devirtualize());
+  pass_manager.add(new smack::MemoryAllocationChecker());
```

Comment:
This should not be enabled by default. You should introduce a command line option that enables it. See `StaticUnroll` for an example of how to do that. You should also add a command like option into the SMACK python script. Maybe call it `--memory-safety`.